docs/architecture-notes.md

```diff
@@ -1,6 +1,36 @@
 # Notes on architecture decisions
 
-## Current croods model
+## Current framework design
+
+The project is split into 3 main pieces, reflected by the root directory structure.
+These parts are defined by where the code is executed and, therefore, where the state is managed in the client-server model.
+
+* ui - code that runs in the browser. This is where all the user interface is defined, mostly accidental complexity on how to draw screens and interact through the browser.
+* domain-logic - code that is shared between client and server. These are the domain actions (or events depending on terminology) that will always run on the server but have proxy functions to be called in the browser. This is where most of our essential complexity lives.
+* jobs - code that runs only on server, either on a clock or upon receiving asynchronous messages.
+
+Besides the section *Definitions* from the [README](../README.md) the framework is agnostic in relation to how the application and its database are modelled. The developer should be able to chose something closer to an event sourced application with immutable data or something more along the lines of mutable entities.
+
+Our main goal is to remove the friction of having to think in terms of HTTP (which is an implementation detail in most systems) or any other protocol and allow the developer to focus on writing the domain logic, which is usually the main concern of the software development process.
+
+Another project goal is to keep interfaces easy to design and enforced on the type level. This should help the developer to code against well defined interfaces reducing bugs and cognitive burden.
+
+### Guiding Principles
```

Comment:
@gustavoguichard special attention to this section. What are your thoughts? Would you remove any of these?

Reply:
Do we really need to keep the "map of paths not taken"? I wouldn't.. but don't mind if you feel like.
I recon we're probably gonna leave a lot more footprint from here on

Reply:
I would like to preserve the tortuous path that got us here. I'll move to another file so we declutter this one

Reply:
But what do you think of the guiding principles?

Reply:
I have extracted the paths not taken to a separate file simplifying the architecture notes and highlighting the path taken!
I have also added short explanation on why we are keeping that document around and why each alternate architecture was abandoned.

Reply:
Awesome! I loved the principles, just no really clear to me what is vertical vs horizontal scale (kind of monorepo vs microservices?)

Thanks for extracting out the "paths not taken".

Reply:
yes, vertical scaling is on the same runtime (or server) while horizontal scaling is when you increase throughput but distributing the runtime.